modules/core/src/main/java/org/apache/ignite/internal/util/OffheapReadWriteLock.java

```diff
@@ -346,7 +347,7 @@ private void signalNextWaiter(int writeWaitCnt, int idx) {
      *      the write lock without leaving a gap. Returns {@code false} otherwise, in this case the resource
      *      state must be re-validated.
      */
-    public Boolean upgradeToWriteLock(long lock, int tag) {
+    public @Nullable Boolean upgradeToWriteLock(long lock, int tag) {
```

Comment:
Seems like `Boolean` could be changed to `boolean`

Reply:
Hm, I've looked a bit closer and it's true that this method can return `null` as a valid outcome, so my comment is incorrect.